util/libcrypto.num

```diff
@@ -5084,10 +5084,43 @@ X509_REQ_set0_signature                 ?	3_0_0	EXIST::FUNCTION:
 X509_REQ_set1_signature_algo            ?	3_0_0	EXIST::FUNCTION:
 OSSL_PARAM_modified                     ?	3_0_0	EXIST::FUNCTION:
 OSSL_PARAM_set_all_unmodified           ?	3_0_0	EXIST::FUNCTION:
+EVP_RAND_fetch                          ?	3_0_0	EXIST::FUNCTION:
+EVP_RAND_up_ref                         ?	3_0_0	EXIST::FUNCTION:
+EVP_RAND_free                           ?	3_0_0	EXIST::FUNCTION:
+EVP_RAND_number                         ?	3_0_0	EXIST::FUNCTION:
+EVP_RAND_name                           ?	3_0_0	EXIST::FUNCTION:
+EVP_RAND_is_a                           ?	3_0_0	EXIST::FUNCTION:
+EVP_RAND_provider                       ?	3_0_0	EXIST::FUNCTION:
+EVP_RAND_get_params                     ?	3_0_0	EXIST::FUNCTION:
+EVP_RAND_CTX_new                        ?	3_0_0	EXIST::FUNCTION:
+EVP_RAND_CTX_free                       ?	3_0_0	EXIST::FUNCTION:
+EVP_RAND_CTX_rand                       ?	3_0_0	EXIST::FUNCTION:
+EVP_RAND_CTX_get_params                 ?	3_0_0	EXIST::FUNCTION:
+EVP_RAND_CTX_set_params                 ?	3_0_0	EXIST::FUNCTION:
+EVP_RAND_gettable_params                ?	3_0_0	EXIST::FUNCTION:
+EVP_RAND_gettable_ctx_params            ?	3_0_0	EXIST::FUNCTION:
+EVP_RAND_settable_ctx_params            ?	3_0_0	EXIST::FUNCTION:
+EVP_RAND_do_all_provided                ?	3_0_0	EXIST::FUNCTION:
+EVP_RAND_names_do_all                   ?	3_0_0	EXIST::FUNCTION:
+EVP_RAND_CTX_instantiate                ?	3_0_0	EXIST::FUNCTION:
+EVP_RAND_CTX_uninstantiate              ?	3_0_0	EXIST::FUNCTION:
+EVP_RAND_CTX_generate                   ?	3_0_0	EXIST::FUNCTION:
+EVP_RAND_CTX_reseed                     ?	3_0_0	EXIST::FUNCTION:
+EVP_RAND_CTX_nonce                      ?	3_0_0	EXIST::FUNCTION:
+EVP_RAND_CTX_set_callbacks              ?	3_0_0	EXIST::FUNCTION:
+EVP_RAND_CTX_enable_locking             ?	3_0_0	EXIST::FUNCTION:
+EVP_RAND_CTX_verify_zeroization         ?	3_0_0	EXIST::FUNCTION:
+EVP_RAND_CTX_strength                   ?	3_0_0	EXIST::FUNCTION:
+EVP_RAND_CTX_state                      ?	3_0_0	EXIST::FUNCTION:
```

Comment:
We don't seem to have  a test for the EVP_RAND_* API functions.

Reply:
The FIPS provider test them as part of the known answer tests plus the DRBG tests do an end to end through them.

I can add explicit tests if these aren't adequate.

Reply:
I think we need explicit tests.

Reply:
Okay, tomorrow.

Reply:
DRBG tests have been added to the EVP test program.
I used NIST's DRBG test data.

Reply:
>I used NIST's DRBG test data.

Great.  I hope there's a comment and a link in the source. Note for giving them credit, but more about future releases looking for updated data.  (I'm sure there is, this PR is just too big for me to skim and confirm that.)

Reply:
Yes, there is a link in the EVP test data file:

```
# Test vectors come from:
# https://csrc.nist.gov/CSRC/media/Projects/Cryptographic-Algorithm-Validation-Program/documents/drbg/drbgtestvectors.zip
```